### PR TITLE
Login Box: Show info message when page directs to login page

### LINKED ISSUE #5456: Login Box: Show info message when page directs to login page
If a user wants to perform different actions, they need to login. However, there is no message usually like "You need to login to perform this action." 

Please identify different cases when the login page shows up and add a message for the user, e.g.
* User verified by clicking verification link show "Thank you for verifying your email address. Please login to continue."
* User clicks "Create Event", but is not logged in show "Please login to create an event. If you have not registered yet, please create an account first. Thank you!"
* What other actions require a user to login? Please add cases and messages.

Screenshot from 2020-11-02 14-36-43

### PR DESCRIPTION
**Pull request by Draco9421**
Screenshot from 2020-11-03 19-08-25

Fixes #5456

#### Changes proposed in this pull request:

- If clicked on create event htrough main page or any event page it shows login alert message if not logged in.

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
> Please identify different cases when the login page shows up and add a message for the user,

This is just 1 case. 2 less than what's even mentioned in the issue as examples.

And will you manually add each case here like you have done for the event?

File: app/templates/login.hbs [hunk 1]
```diff
@@ -1,5 +1,15 @@
-<div class="ui one column container stackable doubling centered grid">
+<div class="ui one column container stackable doubling centered grid text-center">
   <div class="column">
-    <Forms::LoginForm />
+    {{#if (not-eq this.session.Passage "/login")}}
+      {{#if (or (eq this.session.previousRouteName '/') (eq this.session.previousRouteName '/create') (eq this.session.Passage '1'))}}
```
Comment (iamareebjamal) [inline app/templates/login.hbs:4]: Will you add hundreds of cases here like this?

File: app/routes/application.js [hunk 2]
```diff
@@ -130,6 +130,18 @@ export default class ApplicationRoute extends Route.extend(ApplicationRouteMixin
...
+        {
+          this.set('session.Passage', '1')
+        } 
+        else 
+        {
+          this.set('session.Passage', '1')
+        }
+        
+      } else {
+        this.set('session.Passage', '/login');
```
Comment (iamareebjamal) [inline app/routes/application.js:144]: What does this even do?

File: app/templates/login.hbs (hunk 1 above)
Comment (Draco9421) [reply app/templates/login.hbs:4]: No, but if we need different alert message for different case then I have to make cases for them.

File: app/routes/application.js (hunk 2 above)
Comment (Draco9421) [reply app/routes/application.js:144]: I used it to see that whether someone is trying to login through Login page or he/she is redirected to login page.

File: app/templates/login.hbs (hunk 1 above)
Comment (Draco9421) [reply app/templates/login.hbs:4]: I will surely try to generalize the cases.

Comment (Draco9421) [reply app/templates/login.hbs:4]: can I use a action method and check every time that which button is pressed and show the message respectively.

Comment (iamareebjamal) [reply app/templates/login.hbs:4]: No, that is too much work

Comment (Draco9421) [reply app/templates/login.hbs:4]: when we verify our email address it automatically login ,So should I change the message which appears on the screen or should I prevent it from login automatically and show that message.

Comment (iamareebjamal) [reply app/templates/login.hbs:4]: The issue is about messages shown on login screen when you are redirected to them

File: app/routes/application.js (hunk 2 above)
Comment (Draco9421) [reply app/routes/application.js]: Will this method work ?

**Comment by iamareebjamal** [comment]
I don't think your changes are on track of what you're trying to solve. Or your approach is wrong. See how frameworks like flask and django do it. You are extremely complicating it

**Comment by Draco9421** [comment]
Then what approach I should follow ? Any suggestions please.

**Comment by iamareebjamal** [comment]
>   See how frameworks like flask and django do it.

File: app/routes/application.js (hunk 2 above)
Comment (iamareebjamal) [reply app/routes/application.js]: You have written this. How in the world would I know?

**Pull request closed without merging**